weave/ops_arrow/arrow.py

```diff
@@ -182,31 +183,78 @@ def save_instance(self, obj, artifact, name):
                 obj._arrow_data, obj.object_type, obj._artifact, artifact
             )
 
-        from . import convert
+        # v1 AWL format
+        # from . import convert
+
+        # parquet_friendly = convert.to_parquet_friendly(obj)
+        # table = pa.table({"arr": parquet_friendly._arrow_data})
+        # with artifact.new_file(f"{name}.ArrowWeaveList.parquet", binary=True) as f:
+        #     pq.write_table(table, f)
+
+        # v2 AWL format
+        # - if it's a top-level typeddict we don't save in nested sub-array.
+        #   reading specific columns from a parquet or feather file only gives a perf
+        #   improvement for top-level columns.
+        # - saved as feather file instead of parquet. There are some issues in our
+        #   to/from parquet conversion with unions. They can be triggered by using
+        #   v1 format with the hypothesis tests.
```

Comment:
does feather have full support for all arrow formats (e.g., can I directly write a union of dictionary encoded structs)? or do we still need to do massaging for certain exotic arrow arrays?

Reply:
Feather is literally the arrow memory format, but on disk, so it should support all arrow features!